txn.go

```diff
@@ -303,40 +299,46 @@ func (txn *txn) delete(meta TableMeta, guardRevision Revision, data any) (object
 	if table == nil {
 		return object{}, false, tableError(tableName, ErrTableNotLockedForWriting)
 	}
+
 	oldRevision := table.revision
 	table.revision++
-	revision := table.revision
+	newRevision := table.revision
 
 	// Delete from the primary index first to grab the object.
 	// We assume that "data" has only enough defined fields to
 	// compute the primary key.
 	idKey := meta.primary().fromObject(object{data: data}).First()
 	idIndexTree := txn.mustIndexWriteTxn(meta, PrimaryIndexPos)
-	obj, existed := idIndexTree.Delete(idKey)
-	if !existed {
+	var obj object
+	var existed bool
+	if txn.hasDeleteTrackers(meta) {
+		obj, _, existed = idIndexTree.Get(idKey)
```

Comment:
I originally wrote this using `Modify` and also changed part.Modify to support not inserting, but it was still cloning the nodes on the way and it was a bit complex. I didn't see any performance difference between Get+Insert and the Modify so ended up just throwing those changes away.